### PR TITLE
[86by2hktn][d3-chart] tooltip control

### PR DESCRIPTION
**Pull request by msereniti**
## Motivation and Context

In this PR I'm attempted to provide an api to control over d3 chart tooltip position.

I've considered use common React approach when tooltip position is stored in React state, but then preferred imperative api with event's emitter as React state updated might be too slow and requite too much computations to handle popper position.
 
Current api allows developers to control tooltip visibility and position but it require for them to know and update component internal state, so I've split events into more specific ones and exposed only high level `setTooltipVisible` and `setTooltipPosition`. 

I've also found that currently d3 tooltip reinitializes popper instance on every mousemove. To fix it, I also needed to introduce small change into popper.

## How has this been tested?

Manually on all affected chart types.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly or it's not required.
- [x] Unit tests are not broken.
- [x] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

File: semcore/d3-chart/src/Hover.jsx [hunk 1]
```diff
@@ -111,6 +91,7 @@ class HoverLineRoot extends Hover {
 
     return sstyled(styles)(
       <>
+        hi
```
Comment (ilyabrower) [inline semcore/d3-chart/src/Hover.jsx:94]: remove it please)

File: semcore/d3-chart/src/utils.ts [hunk 2]
```diff
@@ -338,3 +333,20 @@ export const calculateBubbleDomain = (
...
+  (event: 'setTooltipVisible', callback: (visible: boolean) => void): Unsubscribe;
+  (event: 'setTooltipPosition', callback: (x: number, y: number) => void): Unsubscribe;
+}
+type PlotEventEmitter = {
+  new (): {
+    emit: PlotEventEmitterEmit;
+    subscribe: PlotEventEmitterSubscribe;
+  };
+};
+export const PlotEventEmitter = EventEmitter as any as PlotEventEmitter;
```
Comment (ilyabrower) [inline semcore/d3-chart/src/utils.ts:352]: What do you think about `EventEmitter<Emit, Subscribe>` ? And some changes in types in utils for this?

File: tools/playground/src/App.tsx [hunk 3]
```diff
@@ -58,13 +58,13 @@ export const PlaygroundWrapper: React.FC = () => {
...
   return (
-    <StrictMode>
-      <div>
-        <Nav />
-        <main className='intergalactic-playground_playground-view'>
-          <PlaygroundWrapper />
-        </main>
-      </div>
-    </StrictMode>
+    // <StrictMode>
```
Comment (ilyabrower) [inline tools/playground/src/App.tsx:61]: turn it back please

File: semcore/popper/src/Popper.jsx [hunk 4]
```diff
@@ -370,15 +370,15 @@ class Popper extends Component {
...
+    const { visible, interaction, disableEnforceFocus, explicitTriggerSet } = this.asProps;
     // @ts-ignore
     const { onKeyDown, ...interactionProps } = this.handlersFromInteraction(
       interaction,
       'trigger',
       visible,
     );
     return {
-      ref: this.createTriggerRef,
+      ref: explicitTriggerSet ? undefined : this.createTriggerRef,
```
Comment (ilyabrower) [inline semcore/popper/src/Popper.jsx:381]: What do you think, if we will use `forkRef(this.asProps.forwardRef, this.createTriggerRef)` ?

And in Tooltip in d3-charts we will create ref and add `getBoundaryRect` method for them

File: semcore/d3-chart/src/Hover.jsx (hunk 1 above)
Comment (msereniti) [reply semcore/d3-chart/src/Hover.jsx:94]: Oh, sorry 🙈

**Review by ilyabrower — APPROVED** [review]

**Pull request merged**